### PR TITLE
Add Op(multinomial) | torchlib(feat)

### PR DESCRIPTION
**Pull request by titaiwangms**
~~1. xfail reason: ONNX spec expects the first dim is batch size, and the whole input should be 2D.~~
~~2. extra opinfo reason: We would like to have a pure 2D datasets to make sure the op is functional.~~

ATen supports 1D and 2D, but ONNX only supports 2D, so Unsqueeze would be used when it's 1D input.

### CODE REVIEW

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 1]
```diff
@@ -1529,6 +1529,19 @@ def _where_input_wrangler(
         matcher=lambda sample: len(sample.args) > 0,
         reason="this ATen overload only supports one tensor as input by design",
     ),
+    TorchLibOpInfo(
+        "multinomial",
+        core_ops.aten_multinomial,
+        nondeterministic=True,
+    ).xfail(
+        matcher=lambda sample: len(sample.input.shape) < 2,
+        reason="onnx spec requires input tensor to have at least 2 dimensions (the first one is batch size)",
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:1538]: Can we use unqueeze and squeeze to handle this?

Comment (titaiwangms) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py]: Done

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 2]
```diff
@@ -1545,6 +1545,11 @@ def _where_input_wrangler(
         matcher=lambda sample: len(sample.args) > 0,
         reason="this ATen overload only supports one tensor as input by design",
     ),
+    TorchLibOpInfo(
+        "multinomial",
+        core_ops.aten_multinomial,
+        nondeterministic=True,
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:1551]:
```suggestion
        nondeterministic=True
```

so the lines can be formatted compactly

**Review by justinchuby — APPROVED** [review]
Lgtm!

**Pull request merged**